### ISSUE TITLE
C code not working when using main function

### ISSUE THREAD

**Issue by Janderhacker** [twibiral/obsidian-execute-code#226]
When I try to run C code without the main function it works:
![image](https://user-images.githubusercontent.com/55597465/218709976-e5fa72e9-848d-4af8-8f17-d20f90c86427.png)

But when I try to use the same code with the main function it stops working:
![image](https://user-images.githubusercontent.com/55597465/218710485-afb869ec-68c5-4f66-82a6-10248c147676.png)

**Comment by milan338** [comment]
Have you enabled the 'use main function' option in the plugin settings for C?

**Comment by Janderhacker** [comment]
Yes, the error in the second image shows up, when I activate the 'use main function' option.

**Comment by milan338** [comment]
Looks like this isn't an issue with the plugin, it's an issue with cling itself, see https://github.com/root-project/cling/issues/364. Until that's fixed, try sticking to not using main, or if you need to, try writing your c code in a c++ block.

I'm not sure when this will be fixed, going forward it might be good to disable the option for now, or only enable this option for remote code execution if / when that gets implemented.

**Comment by benvxavier** [comment]
### 1.  Writing your C code in a C++ block.

![img1](https://user-images.githubusercontent.com/116689016/225865430-273f70c6-c350-4ea6-999b-db0c628af3fd.png)

### 2. Enabled the 'use main function' option in the plugin settings for C++

![img2](https://user-images.githubusercontent.com/116689016/225865448-1be3ee00-c7a8-4c2a-a0da-06ffe5ff45e3.jpeg)

**Comment by benvxavier** [comment]
> ### 1. Writing your C code in a C++ block.
> ![img1](https://user-images.githubusercontent.com/116689016/225865430-273f70c6-c350-4ea6-999b-db0c628af3fd.png)
> 
> ### 2. Enabled the 'use main function' option in the plugin settings for C++
> ![img2](https://user-images.githubusercontent.com/116689016/225865448-1be3ee00-c7a8-4c2a-a0da-06ffe5ff45e3.jpeg)

...OR...

> ### 1. Writing your C code block.
> ### 2. Disable the 'use main function' option in the plugin settings for C
